### ISSUE TITLE
Only enlarge file name column when expanding file browser side panel

### ISSUE THREAD

**Issue by paulsonak** [jupyterlab/jupyterlab#16659]
## Description

In JupyterLab 4.1.8 and below, when you resize the file browser, only the Name column width changes. But in 4.2.X the Name and Last Modifiied/Modified columns both expand equally. This results in difficulty reading long file names since the Modified column takes up ~half the available space.

### 4.1.8 behavior:
<img width="784" alt="Screenshot 2024-08-08 at 3 58 18 PM" src="https://github.com/user-attachments/assets/903255dc-5769-4980-9701-10ddfb18fa1e">

### 4.2.X behavior:
<img width="784" alt="Screenshot 2024-08-08 at 3 59 29 PM" src="https://github.com/user-attachments/assets/1f463310-fbe3-4d5e-9c34-0f27b76179c3">

## Reproduce
1. Expand the file browser on left side
2. Click on the divider between workspace and browser and drag right
3. Note the size of the name column and the modified column both expand

I noticed this because I use the feature all the time to double check which file I need to read into my notebook.

## Expected behavior

Previously, only the name column expanded when dragging outward. I would like this behavior to be restored, or optionally add the ability for column size adjustment inside the file browser.

## Context

- Operating System and version: MacOS arm64 Sonoma
- Browser and version: Chrome Version 127.0.6533.100 (Official Build) (arm64)
- JupyterLab version: 4.2.X

<details><summary>Troubleshoot Output</summary>
<pre>
$PATH:
        /Users/apaulson/atomsci-venv/bin
        /Users/apaulson/atomsci-venv/bin
        /Users/apaulson/.sdkman/candidates/grails/current/bin
        /Library/Frameworks/Python.framework/Versions/3.9/bin
        /opt/homebrew/bin
        /opt/homebrew/sbin
        /usr/local/bin
        /System/Cryptexes/App/usr/bin
        /usr/bin
        /bin
        /usr/sbin
        /sbin
        /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin
        /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin
        /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin
        /opt/X11/bin
        /Applications/VMware Fusion.app/Contents/Public
        /Users/apaulson/.orbstack/bin

sys.path:
        /Users/apaulson/atomsci-venv/bin
        /Library/Frameworks/Python.framework/Versions/3.9/lib/python39.zip
        /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9
        /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload
        /Users/apaulson/atomsci-venv/lib/python3.9/site-packages
        /Users/apaulson/repos/AMPL_umbrella/AMPL

sys.executable:
        /Users/apaulson/atomsci-venv/bin/python3.9

sys.version:
        3.9.12 (v3.9.12:b28265d7e6, Mar 23 2022, 18:22:40) 
        [Clang 13.0.0 (clang-1300.0.29.30)]

platform.platform():
        macOS-14.3.1-arm64-arm-64bit

which -a jupyter:
        /Users/apaulson/atomsci-venv/bin/jupyter
        /Users/apaulson/atomsci-venv/bin/jupyter

pip list:
        Package                      Version        Editable project location
        ---------------------------- -------------- ----------------------------------------
        absl-py                      2.1.0
        aiohttp                      3.9.5
        aiosignal                    1.3.1
        aiosqlite                    0.20.0
        annotated-types              0.7.0
        anyio                        4.4.0
        appnope                      0.1.4
        argon2-cffi                  23.1.0
        argon2-cffi-bindings         21.2.0
        arrow                        1.3.0
        asttokens                    2.4.1
        astunparse                   1.6.3
        async-lru                    2.0.4
        async-timeout                4.0.3
        atomsci-ampl                 1.6.1          /Users/apaulson/repos/AMPL_umbrella/AMPL
        attrs                        23.2.0
        Babel                        2.15.0
        beautifulsoup4               4.12.3
        bleach                       6.1.0
        bokeh                        3.4.2
        bravado                      11.0.3
        bravado-core                 6.1.1
        cachetools                   5.4.0
        certifi                      2024.7.4
        cffi                         1.16.0
        charset-normalizer           3.3.2
        click                        8.1.7
        cloudpickle                  3.0.0
        coloredlogs                  15.0.1
        comm                         0.2.2
        contourpy                    1.2.1
        cycler                       0.12.1
        Cython                       0.29.37
        dask                         2024.8.0
        dataclasses-json             0.6.7
        debugpy                      1.8.2
        decorator                    5.1.1
        deepchem                     2.7.1
        deepmerge                    1.1.1
        defusedxml                   0.7.1
        dgl                          1.1.2
        dgllife                      0.3.2
        dill                         0.3.8
        distributed                  2024.8.0
        exceptiongroup               1.2.2
        executing                    2.0.1
        faiss-cpu                    1.8.0
        fastjsonschema               2.20.0
        filelock                     3.15.4
        flatbuffers                  24.3.25
        fonttools                    4.53.1
        fqdn                         1.5.1
        frozenlist                   1.4.1
        fsspec                       2024.6.1
        future                       1.0.0
        gast                         0.6.0
        google-auth                  2.32.0
        google-auth-oauthlib         1.0.0
        google-pasta                 0.2.0
        grpcio                       1.65.1
        h11                          0.14.0
        h5py                         3.11.0
        httpcore                     1.0.5
        httpx                        0.27.0
        humanfriendly                10.0
        hyperopt                     0.2.7
        idna                         3.7
        importlib_metadata           8.1.0
        importlib_resources          6.4.0
        iniconfig                    2.0.0
        ipykernel                    6.29.5
        ipython                      8.18.1
        ipywidgets                   8.1.3
        isoduration                  20.11.0
        jedi                         0.19.1
        Jinja2                       3.1.4
        joblib                       1.4.2
        json5                        0.9.25
        jsonpatch                    1.33
        jsonpath-ng                  1.6.1
        jsonpointer                  3.0.0
        jsonref                      1.1.0
        jsonschema                   4.23.0
        jsonschema-specifications    2023.12.1
        jupyter_ai                   2.20.0
        jupyter_ai_magics            2.20.0
        jupyter_client               8.6.2
        jupyter_core                 5.7.2
        jupyter-events               0.10.0
        jupyter-lsp                  2.2.5
        jupyter_server               2.14.2
        jupyter_server_terminals     0.5.3
        jupyterlab                   4.2.4
        jupyterlab_pygments          0.3.0
        jupyterlab_server            2.27.3
        jupyterlab_widgets           3.0.11
        keras                        2.14.0
        kiwisolver                   1.4.5
        langchain                    0.2.12
        langchain-community          0.2.11
        langchain-core               0.2.29
        langchain-text-splitters     0.2.2
        langsmith                    0.1.98
        libclang                     18.1.1
        lightning                    2.3.3
        lightning-utilities          0.11.6
        llvmlite                     0.43.0
        locket                       1.0.0
        maestrowf                    1.1.10
        Markdown                     3.6
        markdown-it-py               3.0.0
        MarkupSafe                   2.1.5
        marshmallow                  3.21.3
        matplotlib                   3.9.1
        matplotlib-inline            0.1.7
        matplotlib-venn              0.11.10
        mdurl                        0.1.2
        mistune                      3.0.2
        ml-dtypes                    0.2.0
        MolVS                        0.1.1
        monotonic                    1.6
        mordred                      1.2.0
        mpmath                       1.3.0
        msgpack                      1.0.8
        multidict                    6.0.5
        mypy-extensions              1.0.0
        nbclient                     0.10.0
        nbconvert                    7.16.4
        nbformat                     5.10.4
        nest-asyncio                 1.6.0
        networkx                     2.8.8
        notebook                     7.2.1
        notebook_shim                0.2.4
        numba                        0.60.0
        numpy                        1.24.4
        oauthlib                     3.2.2
        opt-einsum                   3.3.0
        orjson                       3.10.6
        overrides                    7.7.0
        packaging                    24.1
        pandas                       2.2.2
        pandocfilters                1.5.1
        parso                        0.8.4
        partd                        1.4.2
        pexpect                      4.9.0
        pillow                       10.4.0
        pip                          24.1.2
        platformdirs                 4.2.2
        pluggy                       1.5.0
        ply                          3.11
        prometheus_client            0.20.0
        prompt_toolkit               3.0.47
        protobuf                     4.25.3
        psutil                       6.0.0
        ptyprocess                   0.7.0
        pure_eval                    0.2.3
        py4j                         0.10.9.7
        pyarrow                      17.0.0
        pyasn1                       0.6.0
        pyasn1_modules               0.4.0
        pycparser                    2.22
        pydantic                     2.8.2
        pydantic_core                2.20.1
        Pygments                     2.18.0
        pynndescent                  0.5.13
        pyparsing                    3.1.2
        pytest                       8.3.1
        python-dateutil              2.9.0.post0
        python-json-logger           2.0.7
        pytorch-lightning            2.3.3
        pytz                         2024.1
        PyYAML                       5.4.1
        pyzmq                        26.0.3
        rdkit                        2023.3.3
        referencing                  0.35.1
        requests                     2.32.3
        requests-oauthlib            2.0.0
        rfc3339-validator            0.1.4
        rfc3986-validator            0.1.1
        rich                         13.7.1
        rpds-py                      0.19.0
        rsa                          4.9
        scikit-learn                 1.2.2
        scipy                        1.8.1
        seaborn                      0.13.2
        Send2Trash                   1.8.3
        setuptools                   58.1.0
        simplejson                   3.19.2
        six                          1.16.0
        sniffio                      1.3.1
        sortedcontainers             2.4.0
        soupsieve                    2.5
        SQLAlchemy                   2.0.32
        stack-data                   0.6.3
        supervenn                    0.5.0
        swagger-spec-validator       3.0.4
        sympy                        1.13.1
        tabulate                     0.9.0
        tblib                        3.0.0
        tenacity                     8.5.0
        tensorboard                  2.14.1
        tensorboard-data-server      0.7.2
        tensorflow                   2.14.0
        tensorflow-estimator         2.14.0
        tensorflow-io-gcs-filesystem 0.37.1
        tensorflow-macos             2.14.0
        termcolor                    2.4.0
        terminado                    0.18.1
        threadpoolctl                3.5.0
        tinycss2                     1.3.0
        tomli                        2.0.1
        toolz                        0.12.1
        torch                        2.0.1
        torch_geometric              2.5.3
        torchmetrics                 1.4.0.post0
        tornado                      6.4.1
        tqdm                         4.66.4
        traitlets                    5.14.3
        types-python-dateutil        2.9.0.20240316
        typing_extensions            4.12.2
        typing-inspect               0.9.0
        tzdata                       2024.1
        umap-learn                   0.5.6
        uri-template                 1.3.0
        urllib3                      2.2.2
        wcwidth                      0.2.13
        webcolors                    24.6.0
        webencodings                 0.5.1
        websocket-client             1.8.0
        Werkzeug                     3.0.3
        wget                         3.2
        wheel                        0.43.0
        widgetsnbextension           4.0.11
        wrapt                        1.14.1
        xgboost                      2.1.0
        xyzservices                  2024.6.0
        yarl                         1.9.4
        zict                         3.0.0
        zipp                         3.19.2
</pre>
</details>

<details><summary>Command Line Output (while resizing file browser)</summary>
<pre>
[D 2024-08-08 16:16:46.213 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:16:46.216 LabApp] 204 PUT /lab/api/workspaces/auto-b?1723159006208 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 3.88ms
[D 2024-08-08 16:16:47.792 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:16:47.834 ServerApp] 200 GET /api/contents?content=1&hash=0&1723159007787 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 43.09ms
[D 2024-08-08 16:16:53.242 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:16:53.245 LabApp] 204 PUT /lab/api/workspaces/auto-b?1723159013237 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 3.60ms
[D 2024-08-08 16:16:54.606 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:16:54.607 TerminalsExtensionApp] 200 GET /api/terminals?1723159014601 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 2.32ms
[D 2024-08-08 16:16:54.612 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:16:54.618 LabApp] 200 GET /lab/api/workspaces?1723159014609 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 7.37ms
[D 2024-08-08 16:16:54.621 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:16:54.622 ServerApp] 200 GET /api/sessions?1723159014613 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 2.13ms
[D 2024-08-08 16:16:54.623 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:16:54.624 ServerApp] 200 GET /api/kernels?1723159014614 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 2.83ms
[D 2024-08-08 16:16:57.850 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:16:57.897 ServerApp] 200 GET /api/contents?content=1&hash=0&1723159017846 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 47.95ms
[D 2024-08-08 16:17:04.620 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:17:04.622 TerminalsExtensionApp] 200 GET /api/terminals?1723159024615 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 2.31ms
[D 2024-08-08 16:17:04.626 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:17:04.632 LabApp] 200 GET /lab/api/workspaces?1723159024622 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 6.48ms
[D 2024-08-08 16:17:04.635 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:17:04.636 ServerApp] 200 GET /api/sessions?1723159024625 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 2.17ms
[D 2024-08-08 16:17:04.636 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:17:04.637 ServerApp] 200 GET /api/kernels?1723159024626 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 3.01ms
[D 2024-08-08 16:17:07.918 ServerApp] Accepting token-authenticated request from ::1
[D 2024-08-08 16:17:07.948 ServerApp] 200 GET /api/contents?content=1&hash=0&1723159027913 (8a64a7db409b4cb8875304b5b35c1e2b@::1) 30.77ms</pre>
</details>

<details><summary>Browser Output (nothing specific happens when I drag)</summary>

<pre>
debug.js:87 (3)(+0040315): Connector_Browser: onFrameLoaded for http://localhost:8888/lab/workspaces/auto-b; http://localhost:8888/lab/workspaces/auto-b

debug.js:87 (3)(+0000025): HTTP GET chrome-extension://ekhagklcjbdpajgpjgmbionohlpdbjgc/utilities/resource/dateFormats.json

debug.js:87 (3)(+0000460): Translators: Looking for translators for http://localhost:8888/lab/workspaces/auto-b

debug.js:87 (3)(+0000001): HTTP GET chrome-extension://ekhagklcjbdpajgpjgmbionohlpdbjgc/utilities/resource/dateFormats.json

debug.js:87 (4)(+0000002): Translate: Binding sandbox to http://localhost:8888/lab/workspaces/auto-b

debug.js:87 (4)(+0000008): Translate: Parsing code for unAPI (e7e01cac-1e37-4da6-b078-a0e8343b0e98, 2019-07-15 19:40:00)

debug.js:87 (4)(+0000004): Translate: Parsing code for COinS (05d07af9-105a-4572-99f6-a8e231c0daef, 2021-06-03 14:25:00)

debug.js:87 (4)(+0000013): Translate: Parsing code for Embedded Metadata (951c027d-74ac-47d4-a107-9c3069ab7b48, 2024-06-14 04:50:00)

debug.js:87 (3)(+0000000): Translate: Embedded Metadata: found 2 meta tags

debug.js:87 (4)(+0000004): Translate: Parsing code for DOI (c159dcfe-8a53-4301-a499-30f6549c340d, 2024-05-17 20:25:00)

debug.js:87 (3)(+0000000): Translate: All translator detect calls and RPC calls complete:

debug.js:87 (3)(+0000000): 	No suitable translators found

debug.js:87 (5)(+0000000): Translate: Running handler 0 for translators

jlab_core.e5c4c4689689f1651d98.js?v=e5c4c4689689f1651d98:1 Language pack 'en_US' not installed!
fetch @ jlab_core.e5c4c4689689f1651d98.js?v=e5c4c4689689f1651d98:1
148.aa84a7d464636aa796fe.js?v=aa84a7d464636aa796fe:1 Creating a new websocket connection for inline completions...
148.aa84a7d464636aa796fe.js?v=aa84a7d464636aa796fe:1 Creating a new websocket connection for chat...
:8888/lab/api/extensions?query&page=1&per_page=30&refresh=0&1723159281024:1 
        
        
       Failed to load resource: the server responded with a status of 500 (Internal Server Error)
debug.js:87 (3)(+0048765): Connector_Browser: onActivated for 

debug.js:87 (3)(+0000049): Connector_Browser: onFrameLoaded for http://localhost:8888/lab/api/extensions?query&page=1&per_page=30&refresh=0&1723159281024; http://localhost:8888/lab/api/extensions?query&page=1&per_page=30&refresh=0&1723159281024

debug.js:87 (3)(+0000018): HTTP GET chrome-extension://ekhagklcjbdpajgpjgmbionohlpdbjgc/utilities/resource/dateFormats.json

debug.js:87 (3)(+0000075): Translators: Looking for translators for http://localhost:8888/lab/api/extensions?query&page=1&per_page=30&refresh=0&1723159281024

debug.js:87 (3)(+0000002): HTTP GET chrome-extension://ekhagklcjbdpajgpjgmbionohlpdbjgc/utilities/resource/dateFormats.json

debug.js:87 (4)(+0000002): Translate: Binding sandbox to http://localhost:8888/lab/api/extensions?query&page=1&per_page=30&refresh=0&1723159281024

debug.js:87 (4)(+0000002): Translate: Parsing code for unAPI (e7e01cac-1e37-4da6-b078-a0e8343b0e98, 2019-07-15 19:40:00)

debug.js:87 (4)(+0000001): Translate: Parsing code for COinS (05d07af9-105a-4572-99f6-a8e231c0daef, 2021-06-03 14:25:00)

debug.js:87 (4)(+0000004): Translate: Parsing code for Embedded Metadata (951c027d-74ac-47d4-a107-9c3069ab7b48, 2024-06-14 04:50:00)

debug.js:87 (3)(+0000000): Translate: Embedded Metadata: found 2 meta tags

debug.js:87 (4)(+0000001): Translate: Parsing code for DOI (c159dcfe-8a53-4301-a499-30f6549c340d, 2024-05-17 20:25:00)

debug.js:87 (3)(+0000000): Translate: All translator detect calls and RPC calls complete:

debug.js:87 (3)(+0000000): 	No suitable translators found

debug.js:87 (5)(+0000000): Translate: Running handler 0 for translators

debug.js:87 (3)(+0005886): Connector_Browser: onActivated for http://localhost:8888/lab/workspaces/auto-b

</pre>
</details>

**Comment by krassowski** [comment]
> But in 4.2.X the Name and Last Modifiied/Modified columns both expand equally

CC @JasonWeill I think this may be a side-effect of  #15948/#16207

> or optionally add the ability for column size adjustment inside the file browser

good news, I just opened a PR earlier this week which implements column resizing: https://github.com/jupyterlab/jupyterlab/pull/16646 - if we can find someone to review it, it may even get included in the first beta of JupyterLab 4.3

> only the name column expanded when dragging outward. I would like this behavior to be restored

This is very good feedback. I wondered how it should be when implementing #16646. In the current version of #16646 the last column expands when expanding outwards (if other columns had a width previously set by dragging). I think this should be changed.

I see why what you propose would be useful - because typically it's the name column which contains the longest content. Implementation-wise we would need to change the width calculation algorithm slightly; I would prefer this to be done in a follow-up PR after #16646 is merged to reduce the diff and complexity of the review.

**Comment by JasonWeill** [comment]
FYI: #16646 has been merged in, so now is a fine time for a follow-up PR.

**Comment by JasonWeill** [comment]
If I add `flex-grow: 1` to the style of both the `jp-DirListing-itemName` and `jp-DirListing-headerItem` classes, in Firefox, I see a layout that provides ample space for both "Name" and "Modified", although it breaks the column resizer, which no longer tracks the cursor very closely:

![resize-cols](https://github.com/user-attachments/assets/04bedcd9-8a59-4236-8676-d5a71e106451)

If I fix the modified column to 80 px (not 60) and change the `:last-child` elements' `flex-grow` to 0, I get a nicer display by default, but I can't use the column resizer handle to make the "modified" column any larger.

<img width="285" alt="image" src="https://github.com/user-attachments/assets/d7de1a72-ee75-4bce-bee8-0c3704772c86">

**Comment by JasonWeill** [comment]
Setting the "modified" column's minimum width to 100px will accommodate the "Modified" column in English, and also avoids having ellipses in any of the compact relative dates.

<img width="291" alt="image" src="https://github.com/user-attachments/assets/a0377259-0cb7-495f-bd21-e7a009611078">

**Comment by krassowski** [comment]
Changing the CSS will not help here. The CSS is only used for initial positioning. We do want to change to grid to fix the other issue so I would not spend time playing with `flex` - as you note any change to the current setup will break resizing.

Instead, to fix the issue a change is likely needed in:

https://github.com/jupyterlab/jupyterlab/blob/f7ca6ad67e178995a70883ae284ecf5a7efd3d85/packages/filebrowser/src/listing.ts#L1099-L1158

which is called on resize:

https://github.com/jupyterlab/jupyterlab/blob/f7ca6ad67e178995a70883ae284ecf5a7efd3d85/packages/filebrowser/src/listing.ts#L1071-L1076

Also, see:

https://github.com/jupyterlab/jupyterlab/blob/f7ca6ad67e178995a70883ae284ecf5a7efd3d85/packages/filebrowser/src/listing.ts#L1166-L1200

Which has a bespoke logic for up-sizing individual columns - the `if (previousSize && size && size > previousSize)` block is relevant.
